source/CIS/cis-4.rst

```diff
@@ -0,0 +1,796 @@
+
+Response
+~~~~~~~~
+
+The function output is the issuer's public key.
+It is serialized as :ref:`CIS-4-PublicKeyEd25519`.
+
+.. _CIS-4-functions-registryMetadata:
+
+``registryMetadata``
+^^^^^^^^^^^^^^^^^^^^
+
+Query the registry's metadata.
+
+Response
+~~~~~~~~
+
+The function output is the issuer's metadata URL, the credential type and schema for the credentials stored in the registry.
+
+It is serialized as the issuer's :ref:`CIS-4-MetadataUrl` (``issuer_metadata``) followed by the credential type of the registry :ref:`CIS-4-CredentialType` (``credential_type``) and the corresponding credential JSON schema reference :ref:`CIS-4-SchemaRef` (``credential_schema``)::
+
+  MetadataResponse := (issuer_metadata: MetadataUrl) (credential_type: CredentialType) (credential_schema: SchemaRef)
+
+.. _CIS-4-functions-registerCredential:
+
+``registerCredential``
+^^^^^^^^^^^^^^^^^^^^^^
+
+Register public data for a new credential.
+
+.. note::
+  This standard does not specify how the issuer is authenticated.
+  Implementations can use various mechanisms.
+  For example, the transaction sender's address is checked against the issuer's account address stored in the contract.
+  Another option is to use ``auxiliary_data`` to implement signature-based authentication mechanism.
+
+Parameter
+~~~~~~~~~
+
+The parameter is credential information that is used to create an entry in the registry.
+
+It is serialized as :ref:`CIS-4-CredentialInfo` (``credential_info``) followed by auxiliary data, which is serialized as 2 bytes to encode the length (``n``) of the vector of keys, followed by this many bytes of data::
+
+  AuxData ::= (n: Byte²) (data: Byteⁿ)
+  RegisterCredentialParameter ::= (credential_info: CredentialInfo) (auxiliary_data: AuxData)
+
+Requirements
+~~~~~~~~~~~~
+
+- The credential registration request MUST fail if the credential ID is already present in the registry.
+- After successful registration, querying the credential by its ID with :ref:`CIS-4-functions-credentialEntry` MUST succeed.
+
+.. _CIS-4-functions-revokeCredentialIssuer:
+
+``revokeCredentialIssuer``
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Revoke a credential by the issuer's request.
+
+.. note::
+  This standard does not specify how the issuer is authenticated.
+  Implementations can use various mechanisms.
+  For example, the transaction sender's address is checked against the issuer's account address stored in the contract.
+  Another option is to use ``auxiliary_data`` to implement signature-based authentication mechanism.
+
+Parameter
+~~~~~~~~~
+
+The parameter is the credential ID :ref:`CIS-4-CredentialHolderId` and an optional string in the UTF-8 encoding that indicates the revocation reason.
+
+It is serialized as :ref:`CIS-4-CredentialHolderId` followed by 1 byte to indicate whether a reason is included.
+If its value is 0, then no reason string is present, if the value is 1 then the bytes corresponding to the reason string follow::
```

Comment:
The auxiliary data part of the parameter is left out here. Is that on purpose?

Reply:
Good catch, `AuxData` should be in the explanation.